src/main/java/com/imaee/propinq/buildings/services/implementations/BuildingService.java

```diff
@@ -0,0 +1,38 @@
+package com.imaee.propinq.buildings.services.implementations;
+
+import com.imaee.propinq.buildings.controllers.responses.BuildingResponse;
+import com.imaee.propinq.buildings.data.models.Building;
+import com.imaee.propinq.buildings.data.repositories.IBuildingRepository;
+import com.imaee.propinq.buildings.mappers.BuildingMapper;
+import com.imaee.propinq.buildings.services.interfaces.IBuildingService;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class BuildingService implements IBuildingService {
+    private final IBuildingRepository buildingRepository;
+
+    public BuildingService( IBuildingRepository buildingRepository) {
+        this.buildingRepository = buildingRepository;
+    }
+
+    @Override
+    public List<BuildingResponse> getBuildings() {
+
+        return buildingRepository.findAll()
+                .stream()
+                .map(BuildingMapper::toBuildingResponse)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public BuildingResponse getBuilding(UUID buildingId) {
+       return BuildingMapper.toBuildingResponse(findBuildingByIdOrThrowException(buildingId));
+    }
+    private Building findBuildingByIdOrThrowException(UUID buildingId) {
+        return buildingRepository.findById(buildingId)
+                .orElseThrow(() -> new NoSuchElementException("Building with id " + buildingId + " does not exist." ));
+    }
```

Comment:
Entiendo que el mensaje va a mostrarse en el front, por lo que está **mal** en temas de seguridad devolver el ID del edificio por más descriptivo que sea: evitemos mostrar al cliente información sensible de la base de datos.

Adicionalmente, movamos el mensaje a una constante. Va a ser más fácil de manipular

```suggestion

    // Al principio de la clase
    private static final String NO_BUILDING_MESSAGE = "Building does not exist";

    private Building findBuildingByIdOrThrowException(UUID buildingId) {
        return buildingRepository.findById(buildingId)
                .orElseThrow(() -> new NoSuchElementException(NO_BUILDING_MESSAGE));
    }
```